main/src/ca/uwaterloo/flix/api/lsp/CodeActionTriggerKind.scala

```diff
@@ -0,0 +1,27 @@
+package ca.uwaterloo.flix.api.lsp
+
+import org.json4s.{JInt, JValue}
+import ca.uwaterloo.flix.util.Result
+import ca.uwaterloo.flix.util.Result.{Err, Ok}
+
+sealed trait CodeActionTriggerKind
+
+object CodeActionTriggerKind {
+  def parse(json: JValue): Result[CodeActionTriggerKind, String] = {
+    def err(v: JValue) =
+      Err(s"Code action trigger kind not recognized as a valid integer: '$v'.")
```

Comment:
May be slightly misleading if we see e.g. `not recognized as a valid integer: '3'.`